Plugins/Tweaks/Tweaks/FixScrollLearningBug.cpp

```diff
@@ -0,0 +1,146 @@
+    
+    auto pSpell = Globals::Rules()->m_pSpellArray->GetSpell(nSpellIndex);
+    if (!pSpell)
+        return false;
+
+    auto pPlayer = pServerExoApp->GetClientObjectByObjectId(pThis->m_idSelf);
+    if (pPlayer && !pPlayer->HasExpansionPack(Globals::Rules()->GetSpellExpansionLevel(nSpellIndex), 1))
+        return false;
+
+    CItemRepository* pItemRepository = nullptr;
+    auto pPossessor = pServerExoApp->GetGameObject(pItem->m_oidPossessor);
+    if (pPossessor) {
+        if (pPossessor->m_nObjectType == Constants::ObjectType::Creature) {
+            pItemRepository = pPossessor->AsNWSCreature()->m_pcItemRepository;
+        } else if (pPossessor->m_nObjectType == Constants::ObjectType::Item) {
+            pItemRepository = pPossessor->AsNWSItem()->m_pItemRepository;
+        }
+    }
+
+    if (!pItemRepository) //Not present in decompiled code
+        return 0;
+    
+    if (!pItemRepository->GetItemInRepository(pItem, 0)) {
+        pThis->SendFeedbackMessage(0xdc, nullptr, nullptr);
+        return 0;
+    }
+
+    /*if (pThis->m_pStats->m_nNumMultiClasses == 0) {
+        pThis->SendFeedbackMessage(0x4f, nullptr, nullptr);
+        return 0;
+    }*/
+
+    int nFeedback = 0;
+    bool bLearned = false;
+    for (int nMultiClass = 0; nMultiClass < pThis->m_pStats->m_nNumMultiClasses; nMultiClass++) {
+        auto nClass = pThis->m_pStats->GetClass(nMultiClass);
+        if (Globals::Rules()->m_lstClasses[nClass].m_bCanLearnFromScrolls) {
+            auto nSpellLevel = pSpell->GetSpellLevel(nClass);
+            if (nSpellLevel == (uint8_t) -1) {
+                nFeedback = 0xe0;
+                continue;
+            }
+            auto nSpellGain = pThis->m_pStats->GetSpellGainWithBonus(nMultiClass, nSpellLevel);
+            if (nSpellGain == 0) {
+                nFeedback = 0x50;
+                continue;
+            }
+            if (!pThis->m_pStats->GetSpellMinAbilityMet(nMultiClass, nSpellLevel)) {
+                nFeedback = 0x51;
+                continue;
+            }
+            auto nSchool = pThis->m_pStats->GetSchool(nMultiClass);
+            if (nSchool) {
+                int nOppositeSchool = 0;
+                auto pSchoolTable = Globals::Rules()->m_p2DArrays->m_pSpellSchoolTable;
+                auto bFound = pSchoolTable->GetINTEntry(nSchool, "Opposition", &nOppositeSchool);
+                if (bFound && pSpell->m_nSchool == nOppositeSchool) {
+                    nFeedback = 0xdb;
+                    continue;
+                }
+            }
+
+            auto nKnownSpells = pThis->m_pStats->GetNumberKnownSpells(nMultiClass, nSpellLevel);
+            bool bKnownSpell = false;
+            for (int i = 0; i < nKnownSpells && !bKnownSpell; i++) {
+                if (pThis->m_pStats->GetKnownSpell(nMultiClass, nSpellLevel, i) == nSpellIndex) {
+                    nFeedback = 0xdd;
+                    bKnownSpell = true;
```

Comment:
`break;` here

Reply:
I omitted the break there because I used the `!bKnownSpell` condition. If a break is preferred for legibility/consistency I can change it to a break.

Reply:
Ah, I missed the `!bKnownSpell` case. I think `break;` is much more readable, it's a very common idiom to do:
```c
for (i = 0; i < SOME_MAX; i++) {
   if (something[i] == search) {
       do_found_stuff(i);
       break; // bail early
    }
}
```